app/src/main/java/com/larryhsiao/nyx/jots/JotById.java

```diff
@@ -0,0 +1,45 @@
+package com.larryhsiao.nyx.jots;
+
+import com.silverhetch.clotho.Source;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Calendar;
+
+/**
+ * Source to build jot by Id.
+ */
+public class JotById implements Source<Jot> {
+    private final Long id;
+    private final Source<Connection> db;
+
+    public JotById(Long id, Source<Connection> db) {
+        this.id = id;
+        this.db = db;
+    }
+
+    @Override
+    public Jot value() {
+        try (PreparedStatement stmt = db.value().prepareStatement(
+                // language=H2
+                "SELECT * FROM jots WHERE id=?;"
+        )) {
+            stmt.setLong(1, id);
+            ResultSet res = stmt.executeQuery();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Always check the return of one of the navigation method (next,previous,first,last) of a ResultSet.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5029215)